ovos_utils/gui.py

```diff
@@ -9,6 +13,347 @@ def is_gui_installed():
+
+    @staticmethod
+    def is_gui_installed():
+        return is_gui_installed()
+
+    def is_gui_connected(self):
+        return is_gui_connected(self.bus)
+
+    @staticmethod
+    def can_display():
+        return can_display()
+
+    def is_displaying(self):
+        return self.active_skill is not None
+
+    def is_idle(self):
+        return self._is_idle
+
+    @property
+    def active_skill(self):
+        return self._active_skill
+
+    @property
+    def gui_values(self):
+        return self._datastore
+
+    @property
+    def idle_screens(self):
+        return self._idle_screens
+
+    @property
+    def active_namespaces(self):
+        return self._active_namespaces
+
+    @property
+    def gui_pages(self):
+        return self._loaded
+
+    # GUI event handlers
+    # user can/should subclass this
+    def on_idle(self, namespace):
+        pass
+
+    def on_active(self, namespace):
+        pass
+
+    def on_new_page(self, page, namespace, index):
+        pass
+
+    def on_gui_value(self, namespace, key, value):
+        pass
+
+    ######################################################################
+    # GUI client API
+    # TODO see how much of this can be removed
+    @staticmethod
+    def _get_page_data(message):
+        """ Extract page related data from a message.
+
+        Args:
+            message: messagebus message object
+        Returns:
+            tuple (page, namespace, index)
+        Raises:
+            ValueError if value is missing.
+        """
+        data = message.data
+        # Note:  'page' can be either a string or a list of strings
+        if 'page' not in data:
+            raise ValueError("Page missing in data")
+        if 'index' in data:
+            index = data['index']
+        else:
+            index = 0
+        page = data.get("page", "")
+        namespace = data.get("__from", "")
+        return page, namespace, index
+
+    def _set(self, namespace, name, value):
+        """ Perform the send of the values to the connected GUIs. """
+        if namespace not in self._datastore:
+            self._datastore[namespace] = {}
+        if self._datastore[namespace].get(name) != value:
+            self._datastore[namespace][name] = value
+
+    def __find_namespace(self, namespace):
+        for i, skill in enumerate(self._loaded):
+            if skill[0] == namespace:
+                return i
+        return None
+
+    def __insert_pages(self, namespace, pages):
+        """ Insert pages into the namespace
+
+        Args:
+            namespace (str): Namespace to add to
+            pages (list):    Pages (str) to insert
+        """
+        LOG.debug("Inserting new pages")
+        # Insert the pages into local reprensentation as well.
+        updated = self.Namespace(self._loaded[0].name,
+                                 self._loaded[0].pages + pages)
+        self._loaded[0] = updated
+
+    def __remove_page(self, namespace, pos):
```

Comment:
do we want to add an event users can react to for this?

Reply:
Can't the event for remove pages be used to deduce this info? I guess it could be a convenience but I wouldn't say it's a necessity.

Reply:
its redundant for sure, but since the usage of this is by subclassing feels like i should add it for sake of completeness